Remove irrelevant File and Directory Entries overviews, add comparison for all file APIs (#38061)
mdn/content: files/en-us/web/api/file_and_directory_entries_api/index.md

{{DefaultAPISidebar("File and Directory Entries API")}}

The File and Directory Entries API provides a way to process directories and file lists provided by the user via a form input or a drag-and-drop operation. It is a more advanced version of the [File API](/en-US/docs/Web/API/File), which allows you to work with a single file. It was originally intended to support a full virtual file system, but now only supports read operations on user-provided data.

> This API is unrelated to the [File System API](/en-US/docs/Web/API/File_System_API), which actually provides a virtual file system for web applications to persistently store data. See [File API](/en-US/docs/Web/API/File_API#relationship_to_other_file-related_apis) for a comparison of the three APIs.

Reviewer: ```suggestion
See [Relationship to other file-related APIs](/en-US/docs/Web/API/File_API#relationship_to_other_file-related_apis) for a comparison between this API, the [File System API](/en-US/docs/Web/API/File_System_API), and the [File API](/en-US/docs/Web/API/File_API).
```

As in the other case, consider making this a note.

Author: IMO this is not very note-worthy, since the entrypoints of these two APIs are pretty clear about what they do, it just happens that they are in separate specs and therefore are two different APIs.

## History

The original File System API was created to let browsers implement support for accessing a sandboxed virtual file system on the user's storage device. Work to standardize the specification was abandoned back in 2012, but by that point, Google Chrome included its own implementation of the API. Over time, a number of popular sites and Web applications came to use it, often without providing any means of falling back to standard APIs or even checking to be sure the API is available before using it. Mozilla instead opted to implement other APIs which can be used to solve many of the same problems, such as [IndexedDB](/en-US/docs/Web/API/IndexedDB_API); see the blog post [Why no FileSystem API in Firefox?](https://hacks.mozilla.org/2012/07/why-no-filesystem-api-in-firefox/) for more insights.

> This has caused a number of popular websites not to work properly on browsers other than Chrome. Because of that, an attempt was made to create a spec offering the features of Google's API which consensus could be reached on. The result was the [File and Directory Entries API](/en-US/docs/Web/API/File_and_Directory_Entries_API). This subset of the API provided by Chrome is still not fully specified; however, for web compatibility reasons, it was decided to implement a subset of the API in Firefox; this was introduced in Firefox 50.

Reviewer: ```suggestion
As a result, a number of popular websites did not work properly on browsers other than Chrome. To resolve that, the features of Google's API for which consensus could be reached was standardized as the File and Directory Entries API, and this was then implemented in other browsers.
```
